libs/evcs-service/src/test/java/uk/gov/di/ipv/core/library/pact/ContractTest.java

```diff
@@ -0,0 +1,430 @@
+                                "Authorization",
+                                "Bearer " + TEST_EVCS_ACCESS_TOKEN))
+                .willRespondWith()
+                .status(200)
+                .body(getResponseBodyForUserVC())
+                .toPact();
+    }
+
+    private DslPart getResponseBodyForUserVC() {
+        return newJsonBody(
+                        body -> {
+                            body.array(
+                                    "vcs",
+                                    vcs -> {
+                                        vcs.object(
+                                                vc -> {
+                                                    vc.stringType("vc", "vc");
+                                                    vc.stringType("state", "ABANDONED");
+                                                    vc.object(
+                                                            "metadata",
+                                                            metadata -> {
+                                                                metadata.stringType(
+                                                                        "reason",
+                                                                        "abandoned due inactivity");
+                                                                metadata.numberType(
+                                                                        "timestampMs",
+                                                                        1711721297123L);
+                                                                metadata.stringType(
+                                                                        "txmaEventId",
+                                                                        "1a116fe7-2ff9-4f7c-940d-d55fa7d03d66");
+                                                            });
+                                                });
+                                    });
+                        })
+                .build();
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "validRetrieveVcRequestReturnsUsersVcWith200")
+    void testRetrieveVcRequestReturnsUsersVcWith200(MockServer mockServer) throws Exception {
+        // Under Test
+        EvcsClient evcsClient = new EvcsClient(mockConfigService);
+        var evcsGetUserVCsDto =
+                evcsClient.getUserVcs(TEST_USER_ID, TEST_EVCS_ACCESS_TOKEN, VC_STATES_FOR_QUERY);
+
+        // Assertions
+        assertEquals(1, evcsGetUserVCsDto.vcs().size());
+    }
+
+    @Pact(provider = "EvcsProvider", consumer = "EvcsConsumer")
+    public RequestResponsePact invalidRetrieveVcRequestReturnsWith400(PactDslWithProvider builder) {
+        return builder.given("EVCS client exist")
+                .given("VCS exist")
+                .given("EVC has users VC")
+                .uponReceiving("A request to get VCS")
+                .path("/vcs/" + INVALID_USER_ID)
```

Comment:
Would you expect an invalid user id to give a 400? I'd expect that to be a 404...

You might get a 400 with an invalid state, for example.

Reply:
hm yes makes sense thanks